lib/hanami/environment.rb

```diff
@@ -487,10 +492,11 @@ def set_hanami_env_vars!
     # @since 0.2.0
     # @api private
     def set_application_env_vars!
-      dotenv = root.join(DEFAULT_DOTENV_ENV % environment)
-      return unless dotenv.exist?
-
-      env.load!(dotenv)
+      DEFAULT_DOTENV_ENV_FILES.each do |filename_format|
+        next if filename_format == '.env.local' && environment == 'test'
```

Comment:
I really don't like this. This method could also look like:

```ruby
    def set_application_env_vars!
      [
        root.join(".env.#{environment}.local"),
        (root.join(".env.local") unless environment == 'test'),
        root.join(".env.#{environment}"),
        root.join(".env")
      ].compact.each do |path|
        env.load!(path) if path.exist?
      end
    end
```

But that gets away from the use of constants which I was trying to adhere to.

Reply:
The second line will produce a `nil` value if `environment == "test"` is true. That means `path.exist?` will crash.

Reply:
Yeah, you are right. I would to need to do `path &&  path.exist?`. Do you prefer this way of doing it?

Reply:
Yes, but why do we need to exclude `.env.local` for `test` environment?

Reply:
@gmanley Initially I missed the call to `Array#compact` which resolves that problem. However, `path & path.exist?` allocates less memory: https://gist.github.com/jodosha/4c6629f870a5d78aec8116e301140447

Reply:
@gmanley I'm fine to keep the code as it is with `DEFAULT_DOTENV_ENV_PATTERNS`, but we should extract a private method for that special check for test env.

Something like:

```ruby
next unless dotenv_applicable?(filename_format)
# ...

# @api private
# @see https://github.com/bkeepers/dotenv#what-other-env-files-can-i-use
def dotenv_applicable?(filename_format)
  # ...
end
```

Reply:
Addressed here: https://github.com/hanami/hanami/pull/1004/commits/6dcc93417abc81f6d6bf2de729d0581dfb50392a